scio-parquet/src/main/java/com/spotify/scio/parquet/WriterUtils.java

```diff
@@ -28,12 +31,47 @@ public static <T, SELF extends ParquetWriter.Builder<T, SELF>> ParquetWriter<T>
       ParquetWriter.Builder<T, SELF> builder, Configuration conf, CompressionCodecName compression)
       throws IOException {
     // https://github.com/apache/parquet-mr/tree/master/parquet-hadoop#class-parquetoutputformat
-    int rowGroupSize =
-        conf.getInt(ParquetOutputFormat.BLOCK_SIZE, ParquetWriter.DEFAULT_BLOCK_SIZE);
+    long rowGroupSize =
+        conf.getLong(ParquetOutputFormat.BLOCK_SIZE, ParquetWriter.DEFAULT_BLOCK_SIZE);
+
+    for (Map.Entry<String, Boolean> entry :
+        getColumnarConfig(conf, ParquetOutputFormat.BLOOM_FILTER_ENABLED, Boolean::parseBoolean)
+            .entrySet()) {
+      builder = builder.withBloomFilterEnabled(entry.getKey(), entry.getValue());
+    }
+
+    for (Map.Entry<String, Boolean> entry :
+        getColumnarConfig(conf, ParquetOutputFormat.ENABLE_DICTIONARY, Boolean::parseBoolean)
+            .entrySet()) {
+      builder = builder.withDictionaryEncoding(entry.getKey(), entry.getValue());
+    }
+
+    for (Map.Entry<String, Long> entry :
+        getColumnarConfig(conf, ParquetOutputFormat.BLOOM_FILTER_EXPECTED_NDV, Long::parseLong)
+            .entrySet()) {
+      builder = builder.withBloomFilterNDV(entry.getKey(), entry.getValue());
+    }
+
     return builder
         .withConf(conf)
         .withCompressionCodec(compression)
+        .withDictionaryEncoding(ParquetOutputFormat.getEnableDictionary(conf))
+        .withBloomFilterEnabled(ParquetOutputFormat.getBloomFilterEnabled(conf))
+        .withDictionaryPageSize(ParquetOutputFormat.getDictionaryPageSize(conf))
+        .withMaxRowCountForPageSizeCheck(ParquetOutputFormat.getMaxRowCountForPageSizeCheck(conf))
+        .withMinRowCountForPageSizeCheck(ParquetOutputFormat.getMinRowCountForPageSizeCheck(conf))
+        .withValidation(ParquetOutputFormat.getValidation(conf))
         .withRowGroupSize(rowGroupSize)
         .build();
   }
+
+  private static <T> Map<String, T> getColumnarConfig(
```

Comment:
Unfortunately there isn't a public API available in parquet-mr to do this per-column logic for us :(